src/expressions/dataTypes/casting/tryCastToType.ts

```diff
@@ -147,6 +147,17 @@ function createCastingFunction(from: ValueType, to: ValueType): (value: Value) =
 		});
 	}
 
+	// If the two types are equal, no casting is required and we can just return the value as it is.
+	if (from === to) {
```

Comment:
Just out of curiosity does this branch ever get triggered during the test suite for example?

Reply:
Yes